File: eel-core/src/main/scala/io/eels/component/parquet/ParquetLogMute.scala
package io.eels.component.parquet

import java.util.logging.Logger

import scala.util.Try

object ParquetLogMute {

  // the handler is in a static init block so we must force loading of the class before we unapply it, otherwise
  // when the class is later on loaded, it will just add it back
  def apply(): Unit = {
    Try { Class.forName("org.apache.parquet.Log") }
    Try { Class.forName("parquet.Log") }
    for ( pack <- Seq("org.apache.parquet", "parquet") ) {
      Try {
        val logger = Logger.getLogger(pack)
        logger.getHandlers.foreach(logger.removeHandler)
        logger.setUseParentHandlers(false)
      }
    }
  }
}

File: eel-core/src/main/kotlin/io/eels/component/parquet/ParquetLogMute.kt
package io.eels.component.parquet

import java.util.logging.Logger

object ParquetLogMute {

  fun apply() {

    try {
      Class.forName("org.apache.parquet.Log")
    } catch(e: Exception) {
    }

    try {
      Class.forName("parquet.Log")
    } catch(e: Exception) {
    }

    for (pack in listOf("org.apache.parquet", "parquet")) {
      try {
        val logger = Logger.getLogger(pack)
        logger.handlers.forEach { logger.removeHandler(it) }
        logger.useParentHandlers = false
      } catch(e: Exception) {
      }
    }
  }
}